src/Utils.php

```diff
@@ -220,7 +231,10 @@ public static function isHostInNoProxy(string $host, array $noProxyArray): bool
 
         // Strip port if present.
         if (\strpos($host, ':')) {
-            $host = \explode($host, ':', 2)[0];
+            $hostParts = \explode($host, ':', 2);
+            if (false !== $hostParts) {
+                $host = $hostParts[0];
+            }
```

Comment:
So I was trying to have a real good look on this.. because I was really confused in the way that the code (not yours) is written.. 

So after examining this I believe that `explode` function will never throw false.. as we have already checked that the string contains the `:` and it is able to explode. 

I would be happy to live here with a baseline error or a docblock...

Reply:
You are right, and in addition to that from the `explode` documentation:

> If delimiter is an empty string (""), explode will return false.

An `InvalidArgumentException` is thrown on top of the explode, so no this won't happen. 

Fixed in c5687a9d8f67e113e92dd1490c3f51ccf65d5bc6